Include/internal/pycore_obmalloc.h

```diff
@@ -668,6 +668,15 @@ struct _obmalloc_state {
 #if WITH_PYMALLOC_RADIX_TREE
     struct _obmalloc_usage usage;
 #endif
+    // true if the obmalloc state has been initialized.  This must be done
+    // before the malloc/free functions of obmalloc are called and must be
+    // done only once per obmalloc state.  The function _PyMem_init_obmalloc()
+    // does the initialization.
+    bool initialized;
```

Comment:
This is only needed for `obmalloc_state_main` so we not add `int obmalloc_state_main_initialized = 0;` right next to it, and then drop this field?

(If we statically initialized `obmalloc_state_main` then we wouldn't need this at all.)